### DISCUSSION TITLE
自建Tailscale DERP服务教程 - 夏沫花火zzz🌙 (Muska_Ami)のLife

### DISCUSSION THREAD

**Discussion by giscus** [Muska-Ami/muska-ami-blog#73] [General]
# 自建Tailscale DERP服务教程 - 夏沫花火zzz🌙 (Muska_Ami)のLife

准备工作 准备一台 1C1G 以上配置的服务器（内存不足1G编译netns时可能会爆） 服务器能开放stun的端口 UDP/3478 一个域名 (可选)SSL证书  防火墙放行端口iptables/nftables和服务器安全组都要放行以下端口：  UDP/3478 TCP/<PORT> 或 TCP/443（HTTPS模式只能使用443）  <PORT> 请改为一

https://blog.1l1.icu/2024/04/07/zi-jian-tailscale-derp-fu-wu-jiao-cheng/

**Comment by MillerDix** [comment]
您好 请问域名需要解析到机器上吗。我没有用ssl 现在netcheck能看到有机器 但是没有显示延迟（和你最后cnh, cni-hb一样的情况） 应该是没连接成功吧？

**Reply by Muska-Ami** [reply]
肯定需要解析到机器上，不然Tailscale客户端不知道要连哪
我这里没延迟是因为前面的足够快所以Tailscale就不会往下继续检测了

**Reply by MillerDix** [reply]
好的 感谢 我再研究研究

**Comment by BobbyLeung330** [comment]
为什么我按照这个教程做完后，tailscale netcheck找到了自建服务:
```
Report:
	* Time: 2024-12-16T17:21:04.096807095Z
	* UDP: true
	* IPv4: yes,<IP_ADDR>
	* IPv6: no, but OS has support
	* MappingVariesByDestIP: 
	* PortMapping: 
	* CaptivePortal: false
	* Nearest DERP: HKG
	* DERP latency:
		- HongKong: 300µs   (HKG)
```
**但是status查看提示:**
```
# Health check:
#     - Tailscale could not connect to the 'HKG' relay server. Your Internet connection might be down, or the server might be temporarily unavailable.
```

**Reply by BobbyLeung330** [reply]
防火墙全开了

**Reply by Muska-Ami** [reply]
`tailscale netcheck` 只会检查 UDP 的 STUN 端口，你应该检查你的 SSL 配置 以及 TCP 端口是否正常
